src/StringCodeComponent.hpp

```diff
@@ -0,0 +1,34 @@
+#ifndef STRING_CODE_COMPONENT
+#define STRING_CODE_COMPONENT
+
+#include "StringCodeUtilities.hpp" // for functions on stringCodes, look there for documentation about stringCodes
+
+#include <string>
+
+// Base class for all network components which can be configured by string codes
+class StringCodeComponent
```

Comment:
While I like the idea of this godfather :P class (meaning that it's common for basically all NN components classes), I think that the naming is not the best.
Consider there options:
- StringSerializableComponent
- SerializableComponent (I like it best)
- StringableComponent

Reply:
I'll take no 2, because it will stay the same in case we ever change the system to JSON or something.

Reply:
Little ammendment: The class is not really meant as godfather class though, but really just the class of serializable components. If there comes up a reason, we will additionally have either a base class for all network components (which probably then will derive from serializable) or for all classes in the code (which would not derive from it). But at the moment there was no use for that.